src/metatags/handler.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  internalServerError, noContent, notFound, ok,
+} from '@adobe/spacecat-shared-http-utils';
+import { retrieveSiteBySiteId } from '../utils/data-access.js';
+import { getObjectFromKey, getObjectKeysUsingPrefix } from '../utils/s3-utils.js';
+import SeoChecks from './seo-checks.js';
+
+async function fetchAndProcessPageObject(s3Client, bucketName, key, prefix, log) {
+  const object = await getObjectFromKey(s3Client, bucketName, key, log);
+  if (!object?.scrapeResult?.tags || typeof object.scrapeResult.tags !== 'object') {
+    log.error(`No Scraped tags found in S3 ${key} object`);
+    return null;
+  }
+  const pageUrl = key.slice(prefix.length - 1).replace('scrape.json', ''); // Remove the prefix and .json suffix
+  log.info(`Scraped tags for ${pageUrl} : ${JSON.stringify(object.scrapeResult.tags)}`);
+  return {
+    [pageUrl]: {
+      title: object.scrapeResult.tags.title,
+      description: object.scrapeResult.tags.description,
+      h1: object.scrapeResult.tags.h1 || [],
+    },
+  };
+}
+
+export default async function auditMetaTags(message, context) {
+  const { type, url: siteId } = message;
+  const {
+    dataAccess, log, s3Client,
+  } = context;
+
+  try {
+    log.info(`Received ${type} audit request for siteId: ${siteId}`);
+    const site = await retrieveSiteBySiteId(dataAccess, siteId, log);
+    if (!site) {
+      return notFound('Site not found');
+    }
+    if (!site.isLive()) {
+      log.info(`Site ${siteId} is not live`);
+      return ok();
+    }
```

Comment:
Can we combine this to a single return?

```
if (!site || !site.isLive()) {
  log.info(`Site ${siteId} not found or not live`);
  return notFound('Site not found or not live');
}

```

Reply:
In this case, we won't be able to identify what exact error did the audit encountered - if site is not live or it is not found. We need this info in response so that the user knows what exact error we encountered.

Reply:
more explicit guard statements are preferable